File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
       int ele = nums[0];
       int count = 0;
       for(int i = 0 ; i < nums.size() ; i++){
        if(count == 0){
            ele = nums[i];
            count++;
        }else if(ele == nums[i]){
            count +=1;
        }else{
            count-=1;
        }
       }

       int cnt = 0;
       for(int i = 0; i < nums.size() ; i++){
        if(nums[i]==ele) cnt++;
       }

       if(cnt>=nums.size()/2){
        return ele;
       }
       return -1;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ele = nums[0];
        int count = 0;
        for(int i = 0 ; i < nums.length ; i++){
            if(nums[i]==ele){
                count++;
            }else if(count==0){
                count = 0;
                ele = nums[i];
            }else{
                count--;
            }
        }
        int ct2 = 0;
        for(int i = 0 ; i < nums.length ; i++){
            if(nums[i]==ele) ct2++;
        }
        if(ct2>nums.length/2) return ele;
        return -1;
    }
}